aten/src/ATen/native/SparseTensorUtils.cpp

```diff
@@ -108,4 +115,15 @@ Tensor coo_to_csr(const int64_t* indices, int64_t dim, int64_t nnz) {
   return csr;
 }
 
+Tensor zeros_with_indices_like(const Tensor& t) {
+  TORCH_INTERNAL_ASSERT(t.is_sparse());
+  return at::_sparse_coo_tensor_with_dims_and_tensors(
+      t.sparse_dim(),
+      t.dense_dim(),
+      t.sizes(),
+      t._indices().clone(),
+      at::zeros({1}, t._values().options()).expand_as(t._values()),
+      t.options())._coalesced_(t.is_coalesced());
+}
```

Comment:
I suggest the following implementation:
```suggestion
Tensor zeros_with_indices_like(const Tensor& t) {
  auto r = t.detach().clone();
  auto values = (r.is_sparse() ? r._values() : r.values());
  values.zero_();
  return r;
}
```
that enables the function for sparse compressed tensors and is simpler. What do you think?

Reply:
It is too heavy for the actual use case in the backward (too many kernel launches). It should be very lightweight in my opinion.

Reply:
Makes sense. Btw, what are the 3 vs 1 kernel launches? Is "`coo.clone()` is 2 and `zero_()` is 1 vs `_indices().clone()` is 1" correct?